### ISSUE TITLE
Change Request: SM - Plain language triage group names with improved Select list

### ISSUE THREAD

**Issue by janieTC-USDS** [department-of-veterans-affairs/va-mobile-app#9293]
### 1.) Requestor's Name

Janie Tankard Carnock

### 2.) Requestor's Email

jane.carnock@va.gov

### 3.) VA Team

Health

### 4.) What is the name of the project or change?

Messages - plain language care teams with improved Select

### 5.) Please explain what problem you are trying to solve.

Today, the Secure Messaging (SM) tool does not render care team names, referred to as “triage groups”, in plain language for patients despite their central importance in using the tool. These care team names use a host of internal abbreviations, acronyms, and symbols that are indecipherable to patients, creating significant cognitive load when attempting to parse and decide who to send a message to. For example, a patient will see REN PAC 03 WH with no further explanation or support in translating meaning. This reality undermines the patient experience by creating frustration/confusion and leads to greater admin burden for SM Administrators (MHV Coordinators) who spend more time re-routing messages to the correct group.

### 6.) Please explain how this change would work.

- Backend API updates and enhancements to SM Administrative Portal tool to standardize data (beyond the scope of this ticket)
- Enabled by the forthcoming API and SM Portal updates:
    - On VA.gov/VAHB, introduce standardized plain language identifiers (i.e. location, provider/team name, and care type) from Select list options so users can quickly and confidently identify the right team to message. 
    - On VA.gov/VAHB, introduce [opt groups tag ](https://www.w3schools.com/TAgs/tag_optgroup.asp#:~:text=The%20%3Coptgroup%3E%20tag%20is%20used%20to%20group%20related,options%20are%20easier%20to%20handle%20for%20a%20user)to create category labels and group related triage groups by healthcare system in the Select list. 
    - On VA.gov/VAHB, introduce [combo box component from USWDS](https://designsystem.digital.gov/components/combo-box/) to combine select with typehead to make improved names even quicker to find. Note: this is a new component that needs to be introduced with VADS team to jumpstart that process

### 7.) Features Impacted

Secure messaging

### 8.) Please include all relevant documentation links for this change request.

- [Initiative brief with MVP product requirements](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/mhv-to-va.gov/secure-messaging/product/plain-language-TG-initiative/initiative-brief.md)
- [VA.gov collab cycle ticket for this initiative ](https://github.com/department-of-veterans-affairs/va.gov-team/issues/89985)
- July 2024 [Slack thread with VAHB team members design vetting some of the enhancements](https://dsva.slack.com/archives/C018V2JCWRJ/p1720549490614609) with Janie, Jon, Ryan, and Kaitlin 

### User stories

- As a patient, when selecting a triage group to message, I want to see plain language identifiers (i.e. location, doctor name, team name, and group type) on the dropdown Select component so that I can quickly and confidently identify the right team to message. 
- As a patient, I want to be able to identify groups grouped by healthcare system (facility) so that I can more quickly find what I'm looking for. This will be implemented using opt-group.
- As a patient, I want to be able to use typeahead so that I can filter down options presented. This will be implemented on VA.gov using USWDS.
   
#### Before on VA.gov
![image](https://github.com/user-attachments/assets/402a0e13-1e21-41c9-a46d-c754240b0815)

#### After on VA.gov
Note: Final order of identifiers is pending validation from comparative study.
![image](https://github.com/user-attachments/assets/ee819ebb-64ff-48f5-a9f1-53272e95e97e)
![image](https://github.com/user-attachments/assets/a64d04a9-1475-4866-a963-897ac056ef7c)

### Timeline 
 * August 2024
    * OCTO builds out the prototype in DXP. Ro plugs in to help with starting the initial requirements as placeholders.
    * OCTO leading research study comparing current state to 2 options for ordering of the new language for Veterans (i.e., location first or provider first)
*  September-October 2024
    * Based on learnings from the prototype, Ro + team finalize formal requirements, and tech vetting process is started with SM team.
* Mid September: 
    * target Midpoint Review with demo of prototype and handoff to ByLight team to shepherd from here
* October-November 2024: 
    * With start of PI17 on October 16th, SM admin portal development starts for limited production pilot and any additional FE changes.
* Mid November: target Staging Review to **demo solution in VA.gov and VAHB**
* November-December 2024: development is completed, and we launch limited production pilot for VISN 1 users. 

### 9.) Does your team have all the necessary materials (documents, designs, processes, analytics, APIs, etc) in place for Mobile to make these changes right now?

No - forthcoming - targeting staging review in November 2024

### 10.) Has this change been discussed with VAHB POs (Rachel, Ryan, or Don)?

Ryan - yes

### 11.) Has this change been agreed upon by all impacted VA Stakeholders?

Yes

### 12.) What is your ideal due date for this change request?

November 2024

**Comment by timwright12** [comment]
Looks to involve frontend and backend. @StacyB2023 @jperk51 @ala-yna @dumathane can you all please review?

**Comment by StacyB2023** [comment]
@ala-yna I will set up synch for us to review/decide on next steps. cc: @timwright12

**Comment by jperk51** [comment]
The main BE question is where does the new data come from? The existing triage team endpoint (/messaging/health/recipients) provides a name but I'm assuming that's the field we currently have issues with, is that correct?

**Comment by janieTC-USDS** [comment]
@jperk51 Is there a way to add Cle' Diggins (lead developer on this) to this ticket? I noticed I can't tag him.

We are actively building out a "proof of concept" prototype for the technical design of how this would work. I am unclear on if we would be pushing to the same endpoint (with new logic housed on our side to choose the new names when fields exist) or if we would create a new endpoint.

**Comment by jperk51** [comment]
@janieTC-USDS you should be able to tag him via his github username some people don't have their name attached or the same on gitihub

**Comment by jperk51** [comment]
In that case, for now I'll remove the API team label because until it's decided how the upstream is going to provide the new data there's not really any input we can provide. If you want to pull me into any of those conversations though feel free to tag me or add me to the meeting.

**Comment by janieTC-USDS** [comment]
Okay thanks! I am going to check in with him on that.

In the meantime, Cle' is noting that the data is coming from the same table it does now, and the same endpoint will be used.

If the data is present, it will be returned, otherwise it will not and will function as it does today. We can put in a 'type' param if needed to show this is 'v2' or 'v1' if that would help, but in e.g. a 'group type' is required for a 'new' TG structure. If that is not present, it is an 'old' one

**Comment by rtwell** [comment]
we have some questions
1. Because some users will see a combined list of triage groups from different VA facilities (which are not all VISN 1), what is the plan to reconcile those differences? mobile cannot update older apps, so we will need to maintain the unstructured list. 
2. This a UI problem AND a eng problem
3. Does the mixed list (structured vs unstructured) impact the type ahead feature? 
4. The app currently doesn't have Edit Preferences functionality—do we care?
5. "prototype in DXP" what does this mean / is or should the app be included?
6. "Midpoint Review with demo of prototype" should this include the app?
7. What is the default sort order for the triage group list?
8. Type ahead — Ryan will talk with Design @kellylein we can find time next week

**Comment by janieTC-USDS** [comment]
Hey @rtwell ! Here are some initial thoughts and happy to grab sync time too if it's easier to talk through.

1. We have been thinking about that as well for users who will see groups from different VA facilities (aka health care systems, in the language of VA.gov public website). Today, users see a wide hodge podge of differences across VISNs/facilities so in one sense we will only be making it better for some facilities for the time being while others stay the same. Even for pilot/MVP, we will be able to sort all groups into the optgroup structure of nesting under a healthcare system; it's just that those without the underlying data change will still render in a non-plain-language format (status quo) whereas those in the pilot will render in the improved format. See mockup below. In this scenario, Maryland and DC are following the new format whereas Providence groups are still clustered under providence but no changes to the naming itself (note: these are real life examples of Providence groups today).
![image](https://github.com/user-attachments/assets/e41f07d8-e60c-4d0c-89d5-3eb5ad20ceab)


3. noted

4. I am not sure about the typeahead feature but that is a good callout. i.e. is it going to match against the group header or the triage group names themselves, right? cc: @kfink24 

5. Ooh, Edit Preferences in the app. It might not be a bad idea to consider how we at least create a path for users to do this. I wonder if for MVP this is something we could flag proactively that pushes them out in browser to the new Edit Preferences we are building on VA.gov. Users see on average ~15 groups, so we hear it can get overwhelming. The more control and visibility we can give them over turning on and off excess groups is probably a good thing.

6. "Prototype in DXP" - pardon the ambiguity. This reference is to the new version of the legacy SM Admin Portal (DXP) where Cle' is building the underlying changes to support improved data. DXP only refers to that Admin Portal changes and nothing on the Veteran-facing side of things. In concert with the prototype of the admin portal experience, Cle' will be prototyping how the new data inputted in admin portal manifests end to end in the front end client on web for Veteran-facing changes. Ideally we have a VAHB prototype there too, but I am not entirely sure how best to proceed there with a prototype / proof-of-concept (or maybe just clickable prototype to start)

7. Ideally, it would be great if a prototype can include the VAHB app at the point of Midpoint Review. Not sure how feasible that is given timelines and targeting mid September for that...

8. The default sort order triage group is:  
a. triage groups are grouped by healthcare system (facility) and in alphabetized order based on a system grouping;
b. groups within a healthcare system grouping will be in alphabetized order;
c. Exception: any group identified as primary Care should be moved to the top of the list within a health care system (facility) grouping. 
This is buried but documented over in the feature outline: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/mhv-to-va.gov/secure-messaging/product/plain-language-TG-initiative/initiative-brief.md#user-stories.

9. sounds good!



Thanks for the questions, and again  - do let me know if it's easier to chat through live and/or if there are any follow-ups.

**Comment by rtwell** [comment]
1. ahhhh perfect @janieTC-USDS this is exactly what we need—thank you
2. —
3. I will connect with Kaitlin
4. Edit preferences—we _could_ use a web view for this, however there are a couple downsides out of the box: a) users have to login again 🙄 until we fix that, b) there is not any css on VAgov that helps make web view useful (meaning the VAgov header and footer load, there is no Dark Mode support, etc etc). This feels like a decision we should make with real data—do we know how many users use the edit preference feature? side note: I'm making the assumption we do not want MHV preferences to live outside of MHV (we could put them in the settings part of profile in the app)—is this assumption correct?
5. gotcha
6. we can coordinate and see if the timeline works. if not, no biggie. 
7. excellent thank you
8. 🥳

**Comment by rtwell** [comment]
our thinking is 
1. yes, users can still edit their preferences on the web—not great but still an option for folks. to your point, providing a blurb or tooltip or something to let folks know seems reasonable and should be part of our acceptance criteria. 
2. fwiw we could decide to build this full feature into SM in the app, but that comes with the added time and complexity. not out of the question, by any means. however…
3. IIRC (and correct me if I'm wrong @kfink24) long term we're looking to solve this problem through improving the triage list itself so users no longer even need this functionality (AND they see the names that were missing). so there's a is-it-worth-it question here too I think.

**Comment by janieTC-USDS** [comment]
@clediggins-usds

**Comment by janieTC-USDS** [comment]
Sounds good, thanks @rtwell ! And I am tagging in @clediggins-usds who has been leading the web side prototype development work for this. It sounds like @DonMcCaugheyUSDS might be a good point of contact to loop in on how we can prototype with affinity on VAHB. 

For awareness, we are targeting **September 13** for Cle to have a full-stack prototype of how this would work and look on VA.gov. cc: @kfink24.

**Comment by janieTC-USDS** [comment]
Just noting from a sync just now with @kfink24 and @batemapf for visibility with the VAHB team--

We are decomposing the opt-group functionality imagined on this ticket so that it can launch sooner. i.e. This will be a "quicker win" for Veterans that can stand independently from the rest of the plain language group initiative. @batemapf planning to sync with @rtwell on how to get that smaller-sized work prototyped on VAHB while also prepping to prototype the bigger-sized plain language names themselves (the rest of this ticket in its entirety).

Also please note that the typeahead feature was also descoped from this initiative as a nice-to-have/fast-follow versus requirement for piloting plain language triage group names. Sharing for VAHB awareness.

**Comment by batemapf** [comment]
👋 per my convo with @rtwell, I'm specifying the order in which we see this work happening:

1. Now: Group facilities by health systems (the "opt-group" work). For VA.gov, we're basing groupings on what's defined in the VA.gov CMS (aka Drupal).
2. Later: Utilize longer plain-language triage group names. This is dependent on the MHV SM backend team getting their work done first, which we don't expect to complete until winter.
3. Much later: Type ahead.


cc @kfink24

**Comment by DonMcCaugheyUSDS** [comment]
@batemapf We are waiting for the API change before we can review and determine the mobile work required.

**Comment by batemapf** [comment]
👋 happy new year! bringing this one back to life. i believe there is work that is ready to be taken on here so hoping this message can get it rolling.

**background**
originally this ticket covered a whole lot of functionality, including: (1) grouping secure message recipients by health system; (2) utilizing an updated "plain language" format for recipients; and (3) implementing some kind of type ahead / searching functionality. back in the fall we decided to [break this into separate pieces of work](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/9293#issuecomment-2358811596). this would allow the bits of work that could be completed without dependencies on MHV secure message API changes to move forward (specifically the recipient grouping) while waiting for API changes, as described in this [now, next, future plan](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/9293#issuecomment-2374587248). of course, i dropped the ball and never really followed up / pushed, so i think we're in the same place we were in october.

**next step**
i'd like to get started on the recipient grouping work. this has been [implemented](https://github.com/department-of-veterans-affairs/vets-website/pull/33073) on the va.gov side (though it is still behind a flag and not visible to production users) and does not require any API changes. the one thing it _does_ require, however, is access to the [vamc-ehr.json](https://www.va.gov/data/cms/vamc-ehr.json) data that va.gov uses to create the groupings. guessing there is some connection from VAHB to that data since it is likely what VAHB uses to display the "you might be an Oracle/Cerner patient, you might need to go elsewhere" message. but TBD.

@rtwell @DonMcCaugheyUSDS now that i've dumped all that here...what do i do? should i just create a totally new ticket for simplicity? cc @kfink24

**Comment by BeccaTupaj** [comment]
@batemapf and @rtwell Responding to Patrick's message from a couple months ago. If the decision is to break this into 3 separate pieces of work( the now/next/later, plan), then I would recommend closing this ticket and resubmitting 3 separate change requests, so that we can track each piece of work separately. What are your thoughts?

**Comment by rtwell** [comment]
Since we have the answers we need here, I think the mobile app team can create the tickets we need and link to this one…but if that is confusing or adding more work we can disregard as I am def not the GH specialist. We can start with the first piece and go from there.

**Comment by BeccaTupaj** [comment]
This Change Request has been split into 3 epics per @batemapf 's recommendations above and will be tracked in the VA Mobile App Team's project: #10503  #10504  #10505
